app/main.py

```diff
@@ -1,13 +1,26 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error)
+                   for error in errors],
+        "path": file_path,
+        "status": ("failed" if errors else "passed")
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{
+        "errors": [format_linter_error(error)
+                   if error else "" for error in linter_value],
+        "path": linter_key,
+        "status": ("failed" if linter_value else "passed")}
+        for linter_key, linter_value in linter_report.items()]
```

Comment:
you should re-use code, not duplicate the second function.

Reply:
Fixed in last commit

Reply:
@gesiod Are you sure?
<img width="851" alt="image" src="https://github.com/mate-academy/py-linter-formatter/assets/72562232/5fac8055-fc45-4b0a-9e96-7f33771e8526">

Reply:
Sorry, forgot about push.